### ISSUE TITLE
Facing issues with Mappersmith with the combination of confluent-schema-registry package and latest OpenTelemetry instrumentation

### ISSUE THREAD

**Issue by sureshballa** [tulios/mappersmith#225]
Hi,

We are facing an issue for which we would like to get some help or debug insights into this...

We are not using this library directly, however noticed few things when using confluent-schema-registry which internally uses this mappersmith package. Recently we have upgraded our service (that currently utilizes confluent-schema-registry) with the latest OpenTelemetry instrumentation packages and noticed suddenly we are not getting any response from confluent-schema-registry but underlying schema registry service responds with 200 status code. We have debugged confluent-schema-registry and we figured it was not getting a response from its internal component Mappersmith 

```
Mappersmith--onRequestSocketAssigned {
  useSocketConnectionTimeout: false,
  configure: null,
  onRequestWillStart: null,
  onRequestSocketAssigned: null,
  onSocketLookup: null,
  onSocketConnect: null,
  onSocketSecureConnect: null,
  onResponseReadable: null,
  onResponseEnd: null
}
Mappersmith--onSocketLookup {
  err: null,
  address: '10.120.0.6',
  family: 4,
  host: 'kafka-schema.d295e451e8914ad3b7ae1daed3b758ba.cnodes.io'
}
Mappersmith--onSocketConnect undefined
Mappersmith--Socket ready undefined
Mappersmith--onSocketSecureConnect undefined
Mappersmith--Max listeners 100
Mappersmith--Listener count [
  { count: 1, na: 'close' },
  { count: 2, na: 'data' },
  { count: 5, na: 'end' },
  { count: 2, na: 'error' },
  { count: 1, na: 'pause' },
  { count: 1, na: 'resume' },
  { count: 2, na: 'readable' }
]
Mappersmith--Socket data 338
Mappersmith--Resume undefined
Mappersmith--Readable 2 undefined
Mappersmith--readable Url {
  protocol: 'https:',
  slashes: true,
  auth: '<>',
  host: '<>,
  port: '<>',
  hostname: '<>',
  hash: null,
  search: null,
  query: null,
  pathname: '/config/com.subject',
  path: '/config/com.subject',
  href: 'https://kafka-schema.d295e451e8914ad3b7ae1daed3b758ba.cnodes.io:8085/config/com.subject',
  method: 'get',
  headers: {
    'content-type': 'application/vnd.schemaregistry.v1+json',
    traceparent: '00-bc80a8cfe3f39e8ccfa5647c209eec19-84c3477d3608d6be-01'
  }
}
Mappersmith--Resume undefined
Mappersmith--Socket end undefined
Mappersmith--Close { hadError: false } 
```

**Comment by sureshballa** [comment]
This issue is reproducible in node 10 and node 14 versions. But not an issue in node 12

**Comment by sureshballa** [comment]
Hi, we figured out this issue, the introduction of Dynatrace One Agent in our environment has caused this issue

**Comment by alexanderfloh** [comment]
Investigating this on the One Agent side it looks like the issue was caused by conflicting patching by One Agent and Open Telemetry.
There was a fix made in Open Telemetry (https://github.com/open-telemetry/opentelemetry-js/pull/3133), which is shipped in @opentelemetry/sdk-trace-node v1.6.0, that resolves the issue.
